src/Vehicle/FactGroups/VehicleFactGroup.cc

```diff
@@ -0,0 +1,248 @@
+    , _distanceToNextWPFact         (0, _distanceToNextWPFactName,          FactMetaData::valueTypeDouble)
+    , _headingToHomeFact            (0, _headingToHomeFactName,             FactMetaData::valueTypeDouble)
+    , _distanceToGCSFact            (0, _distanceToGCSFactName,             FactMetaData::valueTypeDouble)
+    , _hobbsFact                    (0, _hobbsFactName,                     FactMetaData::valueTypeString)
+    , _throttlePctFact              (0, _throttlePctFactName,               FactMetaData::valueTypeUint16)
+    , _imuTempFact                  (0, _imuTempFactName,                   FactMetaData::valueTypeInt16)
+{
+    _addFact(&_rollFact,                    _rollFactName);
+    _addFact(&_pitchFact,                   _pitchFactName);
+    _addFact(&_headingFact,                 _headingFactName);
+    _addFact(&_rollRateFact,                _rollRateFactName);
+    _addFact(&_pitchRateFact,               _pitchRateFactName);
+    _addFact(&_yawRateFact,                 _yawRateFactName);
+    _addFact(&_groundSpeedFact,             _groundSpeedFactName);
+    _addFact(&_airSpeedFact,                _airSpeedFactName);
+    _addFact(&_airSpeedSetpointFact,        _airSpeedSetpointFactName);
+    _addFact(&_climbRateFact,               _climbRateFactName);
+    _addFact(&_altitudeRelativeFact,        _altitudeRelativeFactName);
+    _addFact(&_altitudeAMSLFact,            _altitudeAMSLFactName);
+    _addFact(&_altitudeAboveTerrFact,       _altitudeAboveTerrFactName);
+    _addFact(&_altitudeTuningFact,          _altitudeTuningFactName);
+    _addFact(&_altitudeTuningSetpointFact,  _altitudeTuningSetpointFactName);
+    _addFact(&_xTrackErrorFact,             _xTrackErrorFactName);
+    _addFact(&_rangeFinderDistFact,         _rangeFinderDistFactName);
+    _addFact(&_flightDistanceFact,          _flightDistanceFactName);
+    _addFact(&_flightTimeFact,              _flightTimeFactName);
+    _addFact(&_distanceToHomeFact,          _distanceToHomeFactName);
+    _addFact(&_timeToHomeFact,              _timeToHomeFactName);
+    _addFact(&_missionItemIndexFact,        _missionItemIndexFactName);
+    _addFact(&_headingToNextWPFact,         _headingToNextWPFactName);
+    _addFact(&_distanceToNextWPFact,        _distanceToNextWPFactName);
+    _addFact(&_headingToHomeFact,           _headingToHomeFactName);
+    _addFact(&_distanceToGCSFact,           _distanceToGCSFactName);
+    _addFact(&_hobbsFact,                   _hobbsFactName);
+    _addFact(&_throttlePctFact,             _throttlePctFactName);
+    _addFact(&_imuTempFact,                 _imuTempFactName);
+
+    _hobbsFact.setRawValue(QVariant(QString("0000:00:00")));
+}
+
+void VehicleFactGroup::handleMessage(Vehicle* vehicle, mavlink_message_t& message)
+{
+    switch (message.msgid) {
+        case MAVLINK_MSG_ID_ATTITUDE:
+            _handleAttitude(vehicle, message);
+            break;
+
```

Comment:
Standard coding style doesn't have blank lines here. The blank lines just add too much extra air in the source for readability. Also we follow Qt code style which lines up ```case``` at same indent as ```switch```. Although that last thing is not super important.